app/src/main/java/ch/epfl/sdp/peakar/user/profile/NewProfileActivity.java

```diff
@@ -193,26 +205,114 @@ private void fillListView() {
         collectionListView.setOnItemClickListener(collectionClicked);
     }
 
+    /**
+     * Setup the challenge list view.
+     * Fill list with challenges the user is enrolled in
+     */
+    private void fillChallengeListView() {
+        // Show correct text if empty
+        findViewById(R.id.add_challenge).setVisibility(View.VISIBLE);
+        ((TextView)findViewById(R.id.profile_empty_text)).setText(R.string.empty_collection);
+
+        ArrayList<NewChallengeItem> items = new ArrayList<>();
+        List<RemotePointsChallenge> challengeList  =
+                AuthService.getInstance().getAuthAccount().getChallenges().stream().map( c -> (RemotePointsChallenge) c).collect(Collectors.toList());
```

Comment:
```suggestion
                displayedAccount.getChallenges().stream().map( c -> (RemotePointsChallenge) c).collect(Collectors.toList());
```
This way this code will work also for the profile of non authenticated users (look at `fillCollectedListView()` for more details).

Reply:
resolved